src/tools/fuzzing.h

```diff
@@ -896,42 +890,54 @@ class TranslateToFuzzReader {
     if (choice < 50) {
       return makeLocalGet(type);
     }
-    if (choice < 60) {
+    if (choice < 60 && canMakeControlFlow) {
       return makeBlock(type);
     }
-    if (choice < 70) {
+    if (choice < 70 && canMakeControlFlow) {
       return makeIf(type);
     }
-    if (choice < 80) {
+    if (choice < 80 && canMakeControlFlow) {
       return makeLoop(type);
     }
-    if (choice < 90) {
+    if (choice < 90 && canMakeControlFlow) {
       return makeBreak(type);
     }
     using Self = TranslateToFuzzReader;
-    auto options = FeatureOptions<Expression* (Self::*)(Type)>()
-                     .add(FeatureSet::MVP,
-                          &Self::makeBlock,
-                          &Self::makeIf,
-                          &Self::makeLoop,
-                          &Self::makeBreak,
-                          &Self::makeCall,
-                          &Self::makeCallIndirect,
-                          &Self::makeLocalGet,
-                          &Self::makeLocalSet,
-                          &Self::makeLoad,
-                          &Self::makeConst,
-                          &Self::makeUnary,
-                          &Self::makeBinary,
-                          &Self::makeSelect,
-                          &Self::makeGlobalGet)
-                     .add(FeatureSet::SIMD, &Self::makeSIMD);
-    if (type == Type::i32 || type == Type::i64) {
+    FeatureOptions<Expression* (Self::*)(Type)> options;
+    options.add(FeatureSet::MVP,
+                &Self::makeLocalGet,
+                &Self::makeLocalSet,
+                &Self::makeConst);
+    if (canMakeControlFlow) {
+      options.add(FeatureSet::MVP,
+                  &Self::makeBlock,
+                  &Self::makeIf,
+                  &Self::makeLoop,
+                  &Self::makeBreak,
+                  &Self::makeCall,
+                  &Self::makeCallIndirect);
```

Comment:
Because the return types of calls have the same restrictions as the return types of blocks and other control flow constructs both when multivalue is enabled and when it is disabled.